sharding/syncer/service.go

```diff
@@ -53,10 +58,24 @@ func (s *Syncer) Stop() error {
 	return nil
 }
 
+// handleServiceErrors manages a goroutine that listens for errors broadcast to
+// this service's error channel. This serves as a final step for error logging
+// and is stopped upon the service shutting down.
+func (s *Syncer) handleServiceErrors() {
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case err := <-s.errChan:
+			log.Error(fmt.Sprint(err))
```

Comment:
This sucks that you have to format the error to a string. Do you want to add any more context to it so its easier to debug? "Sync service error: %v" or something